contracts/staking.tmy/src/staking.tmy.cpp

```diff
@@ -227,23 +232,36 @@ namespace stakingtoken
       asset total_yield = asset(0, SYSTEM_RESOURCE_CURRENCY);
 
       // Iterate through allocations and add yield (if not unstaking)
-      for (auto itr = staking_allocations_table.begin(); itr != staking_allocations_table.end(); itr++)
+      for (auto itr = staking_allocations_table.begin(); itr != staking_allocations_table.end();)
       {
          if (!itr->unstake_requested)
          {
             asset yield = asset(static_cast<int64_t>(itr->tokens_staked.amount * interval_percentage_yield), SYSTEM_RESOURCE_CURRENCY);
-
+      
             staking_allocations_table.modify(itr, eosio::same_payer, [&](auto &row)
             {
                row.tokens_staked += yield;
             });
-
+      
             total_yield += yield;
+            ++itr; // Move to the next element
```

Comment:
`++itr` is in all three conditional blocks so can move it out (after the conditional logic)